packages/flutter_tools/lib/src/ios/xcodeproj.dart

```diff
@@ -221,15 +224,33 @@ XcodeProjectInterpreter get xcodeProjectInterpreter => context.get<XcodeProjectI
 
 /// Interpreter of Xcode projects.
 class XcodeProjectInterpreter {
+  XcodeProjectInterpreter({
+    @required Platform platform,
+    @required ProcessManager processManager,
+    @required Logger logger,
+    @required FileSystem fileSystem,
+    @required AnsiTerminal terminal,
+  }) : _platform = platform,
+       _fileSystem = fileSystem,
+       _terminal = terminal,
+       _logger = logger,
+       _processUtils = ProcessUtils(logger: logger, processManager: processManager);
```

Comment:
We have to be a bit careful with this pattern since tests can directly inject their own `ProcessUtils`. We can:
1. Pass the `ProcessUtils` here instead of the `ProcessManager`.
1. Change this to `(context lookup) ?? ProcessUtils(logger: logger, ...)`
1. Somehow assert here that `ProcessUtils` hasn't been overridden in the context by a test.
1. Get `ProcessUtils` out of the context and instantiate one everywhere where we need one.

Reply:
Longer-term, should we remove injection of processUtils within tests? We have a pretty nice abstraction in the form of the FakeProcessManager, and besides that ProcessManager itself is fairly easily to mock.

Reply:
Yeah. Right now it's only injected in a few places: `android_device_test.dart`, `fuchsia_device_test.dart`, and `web_fs_test.dart` AFAICT.

Reply:
ProcessUtils can now no longer be injected in tests, this has been rebased after that change